### PR TITLE
Ensure tests verify request params

### PR DESCRIPTION
**Pull request by josephwoodward**

### CODE REVIEW

File: api/prometheus/v1/api_test.go [hunk 1]
```diff
@@ -259,7 +279,7 @@ func TestAPIs(t *testing.T) {
 			reqPath:   "/api/v1/query",
 			reqParam: url.Values{
 				"query":   []string{"2"},
-				"time":    []string{testTime.Format(time.RFC3339Nano)},
```
Comment (JosephWoodward) [inline api/prometheus/v1/api_test.go:262]: Tests formatting to currently fail with the following error message:

```
unexpected request parameters: want map[query:[2] time:[1651799194.268618]], got map[query:[2] time:[2022-05-06T02:06:34.268618+01:00]]
```

so I've updated the tests to use `formatTime` for the time being.

**Comment by kakkoyun** [comment]
We've removed 1.16 support. It worths rebasing https://github.com/prometheus/client_golang/pull/1062

**Comment by josephwoodward** [comment]
@kakkoyun Okay, thanks. Hoping to getting around finishing this PR soon.

**Comment by bwplotka** [comment]
Rebased your PR if you don't mind 🙈

**Pull request merged**